### DISCUSSION TITLE
Export jit script

### DISCUSSION THREAD

**Discussion by Zeiny96** [speechbrain/speechbrain#2108] [General]
Can't transform the ECAPA-TDNN speaker-recognition model into torch script
```py
from speechbrain.pretrained import SpeakerRecognition
model = SpeakerRecognition.from_hparams(source="speechbrain/spkrec-ecapa-voxceleb", savedir="pretrained_models/spkrec-ecapa-voxceleb")
model_jit = torch.jit.script(model)
```
I keep receiving various issues whenever I try to solve one I face another is this even supported or not in this model?

**Comment by Adel-Moumen** [comment]
Hello @Zeiny96,

This is not the way to go with SpeechBrain for compiling/jitting a model. You should add a dict with the key `jit` or `compile` with a boolean `True` and give it to your class (see: https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/pretrained/interfaces.py#L200).

In practice it means doing:
```python
run_opt_defaults = {
        "jit": True
    }

model = SpeakerRecognition.from_hparams(source="speechbrain/spkrec-ecapa-voxceleb", savedir="pretrained_models/spkrec-ecapa-voxceleb", run_opts=run_opt_defaults)
``` 

Note1: if you are using the dev branch, please see this PR #2110. I found a small issue that I fixed. 

Note2: The class SpeakerRecognition is not jittable. However, `compile` is working with this class see #2110.

**Reply by Zeiny96** [reply]
Thanks for your fast reply @Adel-Moumen  ,
So after this step how to save the model to disk as a jit script one, usually after converting it to jit.script I would use model.save but looks like this also won't be how thing work here.

**Reply by Adel-Moumen** [reply]
hmmm let me ask @mravanelli and @fpaissan. They might have an answer for you :-)

**Reply by Zeiny96** [reply]
Thanks for your effort @Adel-Moumen,
Okay I will be waiting, as I want to know how to save it, then load and classify using it again.

Thanks
